database/migrations/platform/2024_01_24_000000_update_codenotes_table.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        Schema::table('CodeNotes', function (Blueprint $table) {
+            $table->index('AuthorID');
```

Comment:
To match [upcoming schema changes](https://github.com/RetroAchievements/RAWeb/blob/master/app/Platform/Models/MemoryNote.php#L16-L22) ahead of time:
```php
$table->index('AuthorID', 'memory_notes_user_id_index');
```

Reply:
Is this even necessary? I didn't run the migrate step and the page seems to be working.

Reply:
Index renamed in latest to `memory_notes_user_id_index`.

> Is this even necessary? I didn't run the migrate step and the page seems to be working.

I encountered some performance issues without adding this index. For some users with a lot of code notes, the query added 100ms+, and with the index this was brought down to <5ms.

Reply:
Bleh. I thought that was an aggregated column, not an index. An index makes sense.